packages/near-fast-auth-signer/src/utils/multi-chain/chains/Bitcoin.ts

```diff
@@ -0,0 +1,448 @@
+        `${this.providerUrl}address/${address}/utxo`
+      );
+      const utxos = response.data.map((utxo: any) => {
+        return {
+          txid:   utxo.txid,
+          vout:   utxo.vout,
+          value:  utxo.value,
+          script: utxo.script,
+        };
+      });
+      return utxos;
+    } catch (error) {
+      console.error('Failed to fetch UTXOs:', error);
+      return [];
+    }
+  }
+
+  /**
+   * Fetches the current fee rate from the Bitcoin network.
+   * This method queries the RPC endpoint for fee estimates and returns the fee rate
+   * expected for a transaction to be confirmed within a certain number of blocks.
+   * The confirmation target is set to 6 blocks by default, which is commonly used
+   * for a balance between confirmation time and cost.
+   *
+   * @returns {Promise<number>} A promise that resolves to the fee rate in satoshis per byte.
+   * @throws {Error} Throws an error if the fee rate data for the specified confirmation target is missing.
+   */
+  async fetchFeeRate(confirmationTarget = 6): Promise<number> {
+    const response = await axios.get(`${this.providerUrl}fee-estimates`);
+    if (response.data && response.data[confirmationTarget]) {
+      return response.data[confirmationTarget];
+    }
+    throw new Error(
+      `Fee rate data for ${confirmationTarget} blocks confirmation target is missing in the response`
+    );
+  }
+
+  /**
+   * Fetches a Bitcoin transaction by its ID and constructs a transaction object.
+   * This function retrieves the transaction details from the blockchain using the RPC endpoint,
+   * then parses the input and output data to construct a `bitcoin.Transaction` object.
+   *
+   * @param {string} transactionId - The ID of the transaction to fetch.
+   * @returns {Promise<bitcoin.Transaction>} A promise that resolves to a `bitcoin.Transaction` object representing the fetched transaction.
+   */
+  async fetchTransaction(transactionId: string): Promise<bitcoin.Transaction> {
+    const { data } = await axios.get<Transaction>(
+      `${this.providerUrl}tx/${transactionId}`
+    );
+    const tx = new bitcoin.Transaction();
+
+    tx.version = data.version;
+    tx.locktime = data.locktime;
+
+    data.vin.forEach((vin) => {
+      const txHash = Buffer.from(vin.txid, 'hex').reverse();
+      const { vout } = vin;
+      const { sequence } = vin;
+      const scriptSig = vin.scriptsig
+        ? Buffer.from(vin.scriptsig, 'hex')
+        : undefined;
+      tx.addInput(txHash, vout, sequence, scriptSig);
+    });
+
+    data.vout.forEach((vout) => {
+      const { value } = vout;
+      const scriptPubKey = Buffer.from(vout.scriptpubkey, 'hex');
+      tx.addOutput(scriptPubKey, value);
+    });
+
+    data.vin.forEach((vin, index) => {
+      if (vin.witness && vin.witness.length > 0) {
+        const witness = vin.witness.map((w) => Buffer.from(w, 'hex'));
+        tx.setWitness(index, witness);
+      }
+    });
+
+    return tx;
```

Comment:
so in fast-auth-signer app, if I want hashed transaction, do I use this function and then encode with base64?

Reply:
Why do you need the transaction in the `fast-auth-signer` and encode it with base64?